src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -7090,13 +7091,27 @@ export class StudyViewPageStore
     @observable blockLoading = false;
 
     readonly selectedSamples = remoteData<Sample[]>({
-        await: () => [this.samples],
+        await: () => [this.samples, this.molecularProfiles],
         invoke: () => {
             //fetch samples when there are only filters applied
             if (this.chartsAreFiltered) {
                 if (!this.hasSampleIdentifiersInFilter) {
                     return Promise.resolve([] as Sample[]);
                 }
+                // here we are validating only the molecular profile ids,
+                // but ideally we should validate the entire filters object
+                const invalidMolecularProfiles = findInvalidMolecularProfileIds(
+                    this.filters,
+                    this.molecularProfiles.result
+                );
+
+                if (invalidMolecularProfiles.length > 0) {
+                    return Promise.reject({
+                        detailedErrorMessage: `Invalid molecular profile id(s): ${invalidMolecularProfiles.join(
```

Comment:
Not sure if this is the best solution. `detailedErrorMessage` is a custom field. We may want to have a well defined `Error` object for consistency and always use the same fields everywere.